lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -83,8 +83,8 @@ def remove_ptr_record(ptr)
 
     def do_remove(name, type)
       zone = get_zone(name)
-      if delete_record(zone['id'], name, type)
-        raise Proxy::Dns::Error.new("Failed to remove record #{name} #{type}") unless rectify_zone(zone['name'])
+      unless delete_record(zone['id'], name, type) and update_soa(zone['id'], zone['name']) and rectify_zone(zone['name'])
```

Comment:
This does change the logic a bit. Personally I consider a ```false``` result from ```delete_record``` a success. If it wasn't found, then that's fine. The smart-proxy API does want to raise a NotFound (HTTP 404) though and I'm not fully compliant with that but ```true``` results into a HTTP 200 and that works just as well. However, if you raise an error like this that results into a HTTP 400 and that does break things in the foreman.